gui/common.cpp

```diff
@@ -71,10 +71,10 @@ QString toFilterString(const QMap<QString,QString>& filters, bool addAllSupporte
 QString getDataDir()
 {
     QSettings settings;
-    const QString dataDir = settings.value("DATADIR", QString()).toString();
```

Comment:
why would we want to remove const?

Reply:
Then the question is if we want to have safe code or performance? This code should not be hot.

Reply:
As I said - it's a stupid limitation. I will take a look at the potential hot path one and roll back the others as well as the warning in general.